crates/accelerate/src/sabre/route.rs

```diff
@@ -171,6 +189,18 @@ impl<'a, 'b> RoutingState<'a, 'b> {
 
             // If we reach here, the node is routable.
             self.gate_order.push(node.py_node_id);
+            // Update the qubit depths for the gates in the dag
+            if self.heuristic.depth.is_some() {
+                if let [a, b] = node.qubits[..] {
+                    let qubit_a = a.to_phys(&self.layout).index();
+                    let qubit_b = b.to_phys(&self.layout).index();
+                    let new_depth =
+                        self.qubit_depths[qubit_a].max(self.qubit_depths[qubit_b]) + 1.0;
+                    self.qubit_depths[qubit_a] = new_depth;
+                    self.qubit_depths[qubit_b] = new_depth;
+                }
```

Comment:
Ohh I see, that makes sense. Since directive nodes, like barriers, don’t correspond to a quantum operation in the way a gate does, they wouldn’t impact the circuit depth. I’m not sure of the best way to check if a node is a directive node or not. Do you have any suggestions on how we might identify these nodes effectively?

Reply:
There is a `directive` field in the `DAGNode` struct used by `SabreDAG`: https://github.com/Qiskit/qiskit/blob/main/crates/accelerate/src/sabre/sabre_dag.rs#L26 that should get set if the python object is a directive. If that's true you can ignore the node.